src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/aliasing-xor-mutability.md

```diff
@@ -0,0 +1,48 @@
+---
+minutes: 0
+---
+
+# Aliasing XOR Mutability
+
+We can use the mutual exclusion of `&T` and `&mut T` references for a single value to model some constraints.
+
+```rust,editable
+fn main() {
+pub struct TransactionInterface(/* some kind of interior state */);
+
+pub struct DatabaseConnection {
+    transaction: TransactionInterface,
+}
+
+impl DatabaseConnection {
+    pub fn new() -> Self { Self { transaction: TransactionInterface(/* again, pretend there's some interior state */) } }
+    pub fn get_transaction(&self) -> &TransactionInterface { &self.transaction }
+    pub fn commit(&mut self) {}
+}
+
+pub fn do_something_with_transaction(transaction: &TransactionInterface) {}
+
+let mut db = DatabaseConnection::new();
+let transaction = db.get_transaction();
+do_something_with_transaction(transaction);
+db.commit();
+do_something_with_transaction(transaction); // 🛠️❌
+}
+```
+
+<details>
+
+- This example shows how we can use the "Aliasing XOR Mutability" rule when it comes to shared and mutable references to model safe access to transactions for a database. This is a loose sketch of such a model, and rust database frameworks you use may not look anything like this in practice.
```

Comment:
It might be useful to recall in layman's terms what "Aliasing XOR Mutability" implies, since a lot of folks may not be familiar with that precise terminology.